### ISSUE TITLE
[Docs] v2 Corrections / Feedback

### ISSUE THREAD

**Issue by polarathene** [containous/traefik#4995]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

Tried to follow or make sense of what info the docs provide.

### What did you expect to see?

Consistency, valid documented properties.

### What did you see instead?

Documentation issues/errors.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.0-alpha6
Codename:     faisselle
Go version:   go1.12.6
Built:        2019-06-18T18:25:38Z
OS/Arch:      linux/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

**traefik_static.toml:**

```toml
[entryPoints]
  [entryPoints.web]
    address = ":80"

[providers]
  [providers.docker]
    endpoint = "unix:///var/run/docker.sock"
  [providers.file]
    watch = true
    filename = "/config/traefik_dynamic.toml"
```

**traefik_dynamic.toml:**

```toml
# Testing if I can reference docker service, and a docker service that can reference middleware here
[providers.file]

[http]
  [http.routers]
    [http.routers.router0]
      entryPoints = ["web"]
      service = "docker.whoami_data" # appears to append `_data` where `data` is the parent directory of the compose file.
      rule = "Host(`test.localhost`)"

    [http.middlewares]
      [http.middlewares.bauth.BasicAuth]
        users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/"]
```

**docker-compose.yml:**

```yml
version: '3'

services:
  reverse-proxy:
    image: traefik:v2.0
    command: --api
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock # So that Traefik can listen to the Docker events
      - $PWD/traefik_static.toml:/etc/traefik/traefik.toml
      - $PWD/traefik_dynamic.toml:/config/traefik_dynamic.toml
  whoami:
    image: containous/whoami
    labels:
      - "traefik.http.routers.whoami.rule=Host(`whoami.docker.localhost`)"
      - "traefik.http.routers.whoami.middlewares=file.bauth"
```

---

### Problem - Log level
[Logs](https://docs.traefik.io/v2.0/observability/logs/#log-level) has examples for each option except the log level. It lists the property and it's values, but the property is split by a space, not being familiar enough with TOML I tried with a space anyhow, as well as adopting the other convention of camel casing `logLevel` which wasn't accepted either, [1.7 docs](https://docs.traefik.io/configuration/logs/) seem to have it outside of the `[log]` section(global?) and doing this didn't raise any errors.

### Fix - Log level
Docs should be updated to remove `log ` prefix to the `level` property. Potentially with an example for good measure?

### Problem - Referencing cross-provider

[Docker provider referencing middleware from file provider](https://docs.traefik.io/v2.0/middlewares/overview/#advanced-configuration) shown as: `file@add-foo-prefix`.

When trying this I get the error(though rather than `add-foo-prefix`, mine is `bauth`, I've just swapped it here to match the doc referenced):

```
reverse-proxy_1  | time="2019-06-21T06:28:19Z" level=error msg="middleware \"docker.file@add-foo-prefix\" does not exist" entryPointName=web routerName=docker.whoami
```

Instead I need to follow the other syntax from the [March Medium article on containous blog](https://blog.containo.us/back-to-traefik-2-0-2f9aa17be305): `file.add-foo-prefix`.

### Fix - Referencing cross-provider

The `.` instead of `@` seems to be the correct way to do this.

### Typo, missing backtick

[Router rules](https://docs.traefik.io/v2.0/routing/routers/#rule) under the tip "*Combining Matchers Using Operators and Parenthesis*", OR is missing a backtick.

### Convention - casing

[`entryPoints` vs `EntryPoints`](https://docs.traefik.io/v2.0/routing/entrypoints/)(compare `Configuration Examples` section to `Configuration` snippets). I think I've seen this mixed casing around other places too. I'm not sure if it's intentional, as a new user to both Traefik and TOML, it threw me off a little as I wasn't sure if casing was important or not initially.

---

Also as feedback from a new user, I found it a bit odd when looking for information about configuration (static/dynamic info and file location) that it was only mentioned under `Getting Started` and not under the relevant configuration category. [Specifically the location for `traefik.toml`](https://docs.traefik.io/v2.0/getting-started/configuration-overview/#configuration-file) that I needed to mount in the container for traefik to discover.

Note how the section on the file provider](https://docs.traefik.io/v2.0/providers/file/) links `dynamic configuration` to the [`overview` section](https://docs.traefik.io/v2.0/providers/overview/) which doesn't really touch on dynamic configuration like [`configuration overview` does](https://docs.traefik.io/v2.0/getting-started/configuration-overview/#configuration-file). 

Perhaps the file location isn't appropriate under `Configuration Discovery` but should be relevant for the [`File` static configuration reference](https://docs.traefik.io/v2.0/reference/static-configuration/file/)?

**Comment by SantoDE** [comment]
Hello @polarathene,

thanks a lot for your feedback :) That really helps driving the docs forward!

We will definitely improve the docs before we leave the alpha stage. You're very welcome to contribute some of the changes as well, if you can find some time.

**Comment by ldez** [comment]
Hello,

The current syntax for referencing cross-provider uses `.` but we changed that for the next version (alpha7) to use `@`.

**Comment by polarathene** [comment]
@ldez Ok.. why do the docs show the new syntax before the version is available? 

I see that a recent PR further changes the order from `provider@name` to `name@provider`. I don't mind either way, it's just confusing to try help test v2 when the docs aren't in sync with the current alpha releases.

**Comment by traefiker** [comment]
Closed by #4996.
